src/acvp_build_register.c

```diff
@@ -2810,113 +2809,204 @@ ACVP_RESULT acvp_build_oes(ACVP_CTX *ctx, char **reg) {
-    json_object_set_string(obj, "type", ctx->module_type);
-    json_object_set_string(obj, "vendorUrl", ctx->vendor_url);
-    json_object_set_string(obj, "implementationDescription", ctx->module_desc);
+    json_object_set_string(obj, "name", module->name);
+    json_object_set_string(obj, "version", module->version);
+    json_object_set_string(obj, "type", module->type);
+    json_object_set_string(obj, "vendorUrl", module->vendor->url);
+    json_object_set_string(obj, "description", module->description);
 
     json_array_append_value(reg_arry, val);
-    *reg = json_serialize_to_string_pretty(reg_arry_val);
-    json_value_free(reg_arry_val);
+
+    *reg = json_serialize_to_string_pretty(reg_arry_val, out_len);
+
+    if (reg_arry_val) json_value_free(reg_arry_val);
 
     return ACVP_SUCCESS;
 }
 
 /*
  * This function builds the JSON message to register a vendor with the
  * validating crypto server
+ *
+ * As of right now, we only support one contact with a name, website,
+ * contact name, and contact email.
  */
-ACVP_RESULT acvp_build_vendors(ACVP_CTX *ctx, char **reg) {
-    JSON_Value *reg_arry_val = NULL;
-    JSON_Value *ver_val = NULL;
-    JSON_Object *ver_obj = NULL;
-
-    JSON_Array *reg_arry = NULL, *contact_array = NULL;
+ACVP_RESULT acvp_register_build_vendor(ACVP_CTX *ctx,
+                                       ACVP_VENDOR *vendor,
+                                       char **reg,
+                                       int *out_len) {
+    JSON_Array *reg_arry = NULL, *phone_array = NULL,
+               *address_array = NULL, *email_array = NULL;
+    JSON_Value *reg_arry_val = NULL, *phone_val = NULL,
+               *address_val = NULL, *val = NULL;
+    JSON_Object *phone_obj = NULL, *address_obj = NULL,
+                *obj = NULL;
+    int i = 0;
 
-    JSON_Value *val = NULL;
-    JSON_Object *obj = NULL;
+    if (!ctx) return ACVP_NO_CTX;
 
     /*
-     * As of right now, we only support one contact with a name, website,
-     * contact name, and contact email
+     * Start the registration array
      */
-    JSON_Value *contact_val = NULL;
-    JSON_Object *contact_obj = NULL;
-    JSON_Array *email_array = NULL;
+    reg_arry_val = json_value_init_array();
+    reg_arry = json_array((const JSON_Value *)reg_arry_val);
 
-    if (!ctx) {
-        ACVP_LOG_ERR("No ctx for build vendor");
-        return ACVP_NO_CTX;
+    /* Add the object containing ACVP version */
+    json_array_append_value(reg_arry, acvp_version_json_value());
+
+    val = json_value_init_object();
+    obj = json_value_get_object(val);
+
+    json_object_set_string(obj, "name", vendor->name);
+
+    if (vendor->website) {
+        json_object_set_string(obj, "website", vendor->website);
+    }
+
+    if (vendor->email) {
+        json_object_set_value(obj, "emails", json_value_init_array());
+        email_array = json_object_get_array(obj, "emails");
+
+        json_array_append_string(email_array, vendor->email);
     }
 
+#if 0
```

Comment:
waiting on support for multiple phone numbers ?

Reply:
This stuff that deffed out was originally for doing the creation of the resources. Kept it around for when we end up doing the query, I'll tweak the code here